TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot/Robot.java

```diff
@@ -0,0 +1,110 @@
+package org.firstinspires.ftc.teamcode.Robot;
+
+import org.firstinspires.ftc.teamcode.Robot.systems.MecanumDriveTrain;
+import android.view.View;
+import com.acmerobotics.dashboard.FtcDashboard;
+import com.acmerobotics.dashboard.config.Config;
+import com.qualcomm.hardware.rev.RevHubOrientationOnRobot;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.IMU;
+import com.qualcomm.robotcore.util.ElapsedTime;
+import com.qualcomm.robotcore.util.Range;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.AngularVelocity;
+import org.firstinspires.ftc.robotcore.external.navigation.YawPitchRollAngles;
+
+@Config
+public abstract class Robot {
+    private IMU imu;
+    private double TargetAngle = 0;
+    private double RobotAngle = 0;
+    private double drive;
+    private double slide;
+    private double turn;
+
+    private MecanumDriveTrain driveTrain;
+    private double RobotX, RobotY;
+    private ElapsedTime runtime = new ElapsedTime();
+
+    FtcDashboard dashboard;
+    private double robotHeading;
+    private double lastTimeAngle;
+    private boolean CurrentAlign = true;
+    private boolean DriverOrientationDriveMode = true;
+
+    public double derivativeConstantAngle;
+    public double proportionalConstantAngle;
+
+    public Telemetry telemetry = null;
+
+    private boolean resettingImu = false;
+    private double AutoStartAngle = 0;
```

Comment:
Are all of these needed?

Reply:
To my understanding yes, there maybe a few that could move down into the ITDbot class.

Reply:
Sounds great!